rbi/stdlib/pathname.rbi

```diff
@@ -1040,7 +1040,7 @@ class Pathname < Object
     params(
         p1: String,
     )
-    .returns(String)
+    .returns(Pathname)
```

Comment:
Are you sure it never returns String? Whoever wrote that felt like it sometimes does

Reply:
Looking at the implementation:

```c
return rb_class_new_instance(1, &str, rb_obj_class(self));
```

It seems to always be returning a `Pathname`.